presto-cli/src/main/java/com/facebook/presto/cli/Query.java

```diff
@@ -73,19 +74,27 @@ public Query(StatementClient client)
 
     public void renderOutput(PrintStream out, OutputFormat outputFormat, boolean interactive)
     {
-        Thread clientThread = Thread.currentThread();
-        SignalHandler oldHandler = Signal.handle(SIGINT, signal -> {
+        SignalHandler intHandler = signal -> {
             if (ignoreUserInterrupt.get() || client.isClosed()) {
                 return;
             }
             client.close();
-            clientThread.interrupt();
-        });
+            Thread.currentThread().interrupt();
```

Comment:
Is this going to do the right thing? I would guess that the handler will be invoked by a different thread than the client thread